src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -694,6 +694,40 @@ Copyright (c) .NET Foundation. All rights reserved.
     </ItemGroup>
   </Target>
 
+  <!--
+    ============================================================
+                                        _CreateSingleFileHost
+      Create the single-file host for the publish scenario if app is being published as a self-contained single-file .net5+ app
+    ============================================================
+     -->
+  <UsingTask TaskName="CreateAppHost" AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
+  <Target Name="_CreateSingleFileHost"
+          Inputs="@(IntermediateAssembly);$(SingleFileHostSourcePath)"
+          Outputs="$(SingleFileHostIntermediatePath)"
+          DependsOnTargets="_GetAppHostPaths"
+          Condition="'$(PublishSingleFile)' == 'true' and
+                     '$(SelfContained)' == 'true' and
+                     '$(SingleFileHostSourcePath)' != '' and
+                     '$(_TargetFrameworkVersionWithoutV)' >= '5.0' and
```

Comment:
nit: Consider using [`$([MSBuild]::IsTargetFrameworkCompatible('net5.0', '$(TargetFrameworkVersion)'))`](https://learn.microsoft.com/en-us/visualstudio/msbuild/property-functions?view=vs-2022#msbuild-targetframework-and-targetplatform-functions) instead of the raw TFM check here and in the Condition on line 864.

Reply:
```suggestion
                     $([MSBuild]::IsTargetFrameworkCompatible('net5.0', '$(TargetFrameworkVersion)')) and
```